Report distro info to DB during Kernel Performance tests

dylan-matthew-turner opened the pull request:
## Need

We want to be able to sort by distro in our test results

## Implementation

During the test, /etc/os-release is read and parsed for VERSION_ID and DISTRO_CODENAME, and those are stored in the results database as tags called "distro_version" and "distro_name."

In the event that the file cannot be read, the file is improperly formatted, or the keys are missing, the resulting values for those tags will be "unknown".

## Testing

Ran tests locally on a 9037 and verified that distro_name and distro_version show up in Grafana (will need to remove test data)

dylan-matthew-turner committed:
upload_cyclictest_results: report distro to DB

Distro name and distro version come from /etc/os-release via
DISTRO_CODENAME and VERSION_ID, respectively.

Signed-off-by: Dylan Turner <dylan.turner@ni.com>

amstewart commented on recipes-kernel/kernel-tests/kernel-performance-tests-files/upload_cyclictest_results.py:
```diff
@@ -132,21 +132,42 @@ def get_test_results(path):
+            os_release = os_release_file.read()
+    except:
+        return ('unknown', 'unknown')
+    stmts = os_release.split('\n')
```
I think, if you open in `"r"` mode, you'll be given a TextWrapper object, and then you can just read it with `.readlines()` or something and let the standard libs split lines for you. Does that work?

amstewart commented on recipes-kernel/kernel-tests/kernel-performance-tests-files/upload_cyclictest_results.py:
```diff
@@ -132,21 +132,42 @@ def get_test_results(path):
+        return ('unknown', 'unknown')
+    stmts = os_release.split('\n')
+    pairs = list(map(lambda e: e.split('='), stmts))
+    variables = dict(pairs)
```
Isn't the `map()` already basically a `dict()`? It seems like you should be able to skip the intermediate list here.

amstewart reviewed (changes requested):
LGTM; pending a few minor suggestions.

dylan-matthew-turner commented:
I simplified based on your suggestions and have now run the tests and seen them shown up in Grafana.

During that I noticed one other change necessary which was to strip the newlines as they got reported to Grafana, so I added that change as well.

dylan-matthew-turner commented:
Oops, and also I just had to remove surrounding quotes from the reported name. `"kirkstone"\n` to `"kirkstone"` to `kirkstone`

Should be good to go now.

dylan-matthew-turner commented:
I can rebase and make these all back to one commit if that's preferable

amstewart commented:
Sure.

blueOkiris commented:
Done

dylan-matthew-turner commented:
Done
